indexer/src/utils/filter.test.ts

```diff
@@ -155,3 +154,54 @@ Deno.test("isKakarotTransaction: `to` address matching KAKAROT_ADDRESS", () => {
   const success = isKakarotTransaction(transaction);
   assertEquals(success, true);
 });
+
+Deno.test(
+  "isRevertedWithOutOfResources: true on status reverted and revert reason",
+  () => {
+    const receipt: TransactionReceipt = {
+      executionStatus: "EXECUTION_STATUS_REVERTED",
+      transactionHash: "0x01",
+      transactionIndex: "0x01",
+      actualFee: "0x01",
+      contractAddress: "0x01",
+      l2ToL1Messages: [],
+      events: [],
+      revertReason: "RunResources has no remaining steps",
+    };
+    const success = isRevertedWithOutOfResources(receipt);
+    assertEquals(success, true);
+  },
+);
+
+Deno.test("isRevertedWithOutOfResources: false on status succeeded", () => {
+  const receipt: TransactionReceipt = {
+    executionStatus: "EXECUTION_STATUS_SUCCEEDED",
+    transactionHash: "0x01",
+    transactionIndex: "0x01",
+    actualFee: "0x01",
+    contractAddress: "0x01",
+    l2ToL1Messages: [],
+    events: [],
+    revertReason: "RunResources has no remaining steps",
```

Comment:
I think the exact message is

`Could not reach the end of the program. RunResources has no remaining steps`

Maybe we could test with it entirely to ensure that it works.